### PR TITLE
9810 dealing with cancellations part 3

### PR DESCRIPTION
**Pull request by Siddartha-tech**
Code changes and did code improvement for FSS health check.

### CODE REVIEW

**Review by vijendr13545 — APPROVED** [review]
Looks good to me.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 1]
```diff
@@ -223,6 +232,13 @@ private async Task<string> SelectLatestPublishedDateBatch(List<Products> product
             return uri;
         }
```
Comment (martyn-fewtrell) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:233]: I realise that you have only added to this method but the cyclic complexity has grown with the latest additions and this is probably ready to be refactored to make it more readable/understandable.

Is it possible to refactor this to reduce the complexity?

**Review by martyn-fewtrell — COMMENTED** [review]
Looking good although I think some of the code would now benefit from refactoring.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs (hunk 1 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:233]: Yes refactored the method.

**Review by martyn-fewtrell — APPROVED** [review]
Thanks for the update

**Pull request merged**